internal/ansible/flags/flag.go

```diff
@@ -207,7 +207,7 @@ func (f *Flags) ToManagerOptions(options manager.Options) manager.Options {
 		options.LeaderElectionNamespace = f.LeaderElectionNamespace
 	}
 	if options.LeaderElectionResourceLock == "" {
-		options.LeaderElectionResourceLock = resourcelock.ConfigMapsResourceLock
+		options.LeaderElectionResourceLock = resourcelock.ConfigMapsLeasesResourceLock
```

Comment:
We need to check further to ensure that we can do it see: https://github.com/kubernetes-sigs/controller-runtime/pull/1773. Also, see for a better explantion: https://github.com/kubernetes-sigs/controller-runtime/pull/1903

Reply:
WDYT about we discuss it in a follow up only for these changes. Do we need to do it for the bump?

Reply:
They are required changes for the bump. ConfigMapsResourceLock does not exist for client-go 0.24.0

Reply:
I validated this one that is fine. We cannot bump here leases.
See:
```
	// If you are using `configmaps`/`endpoints` resource lock and want to migrate to "leases",
	// you might do so by migrating to the respective multilock first ("configmapsleases" or "endpointsleases"),
	// which will acquire a leader lock on both resources.
	// After all your users have migrated to the multilock, you can go ahead and migrate to "leases".
	// Please also keep in mind, that users might skip versions of your controller.
```